src/Chapter2.hs

```diff
@@ -842,7 +874,10 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [a] -> [a]
+rotate num lst 
+  | num >= 0  = take (length lst) (drop num (cycle lst))
```

Comment:
That's a correct solution! Though, it does a little bit more work when the given number is big and the list is small. You will cycle through the list a lot. Maybe you can think of a way to optimize this :slightly_smiling_face:

Reply:
Yep, can just rotate `num `mod` (length lst)` times.